test/explain/algoritms/test_gnn_explainer.py

```diff
@@ -0,0 +1,672 @@
+                            explainer_config=explainer_config,
+                            model_config=model_config, edge_attr=None,
+                            batch=None)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('model', [
+    GNN_regression_singleoutput,
+    GNN_regression_multioutput,
+])
+@pytest.mark.parametrize('return_type', return_types_regression)
+def test_gnn_explainer_graph_regression(
+    edge_mask_type,
+    node_mask_type,
+    model,
+    return_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type="model",
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='graph',
+        return_type=return_type,
+        mode="regression",
+    )
+
+    model = model()
+
+    x = torch.randn(8, 3)
+    edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7],
+                               [1, 0, 2, 1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6]])
+    out = model(x, edge_index, None, None)
+
+    if isinstance(model, GNN_regression_multioutput):
+        target_index = 0
+    else:
+        target_index = None
+
+    explainer = GNNExplainer()
+    # try to explain prediction for node 0
+    explanation = explainer(x=x, edge_index=edge_index, model=model,
+                            target=out, target_index=target_index,
+                            explainer_config=explainer_config,
+                            model_config=model_config, edge_attr=None,
+                            batch=None)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('model', [
+    GNN_regression_singleoutput,
+    GNN_regression_multioutput,
+])
+@pytest.mark.parametrize('return_type', return_types_regression)
+@pytest.mark.parametrize("explanation_type", ["model", "phenomenon"])
+def test_gnn_explainer_with_meta_explainer_regression_graph(
+    edge_mask_type,
+    node_mask_type,
+    model,
+    return_type,
+    explanation_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type=explanation_type,
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='graph',
+        return_type=return_type,
+        mode="regression",
+    )
+
+    model = model()
+
+    x = torch.randn(8, 3)
+    edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7],
+                               [1, 0, 2, 1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6]])
+    out = model(x, edge_index, None, None)
```

Comment:
Can we define these at the top of the file are reuse for all tests?

Reply:
yes good point 👍🏽 , will do